www/docs/operations.mdx

```diff
@@ -0,0 +1,248 @@
+Effection from scratch, you should most likely use `main()` as the
+entry point. `run()` is more bare-bones, and should only be used when
+you need more fine grained control.
+
+Using `main()`, our example looks like this:
+
+```javascript
+import { main } from 'effection';
+
+await main(function*() {
+  throw new Error('oh no!');
+});
+```
+
+Instead of needing to catch the error like we did before, `main()`
+will handle it for us, including printing it to the console.
+
+### Composing Operations
+
+Entry points like `run()` and `main()` are used usually once at the very
+beginning of an Effection program, but the easiest and most common way to
+run an operation is to include it in the body of another operation. To do this,
+we use the `yield*` keyword, which is the Effection equivalent of `await`.
+
+For example, the `sleep()` operation pauses execution for a specified duration
+before resuming. We can call it from any operation using `yield*`;
+
+```javascript
+import { main, sleep } from 'effection';
+
+await main(function*() {
+  yield* sleep(1000);
+  console.log("Hello World!");
+});
+```
+
+Or, it can be embeded in an operation which itself is embedded in another
+operation:
+
+```javascript
+import { main, sleep } from 'effection';
+
+function* makeSlow(value) {
+  yield* sleep(1000);
+  return value;
+}
+
+await main(function*() {
+  let text = yield* makeSlow('Hello World!');
+  console.log(text);
+});
+```
+
+There is no limit to the number of operations that can be nested within
+each other, and in fact, composition is so core to how operations work that
+_every operation in Effection eventually boils down to a combination of
+just three primitive operations_: `action()`, `resource()`, and `suspend()`.
+
+### Cancellation and Cleanup
+
+Perhaps the most critical difference between promises and operations is that
+once started, a promise will always run to completion no matter how long that
+takes. Unfortunately, this makes the guarantees of structured concurrency
+impossible to provide automatically. Operations, on the other hand, can be
+interrupted at any time which means that we never have to worry about them
+staying longer than they are absolutely needed.
+
+Let's consider a hypothetical sleep function implemented using `async/await`. It
+uses `setTimeout()` to resolve a promise once the sleep time has elapsed.
+
+```js
+async function sleep(duration) {
+  await new Promise(resolve => setTimeout(resolve, duration));
+}
+```
+
+We can now write our slow hello world program using this version:
+
+```js
+async function main() {
+  await sleep(1000);
+  console.log("Hello World!");
+}
+
+await main();
+```
+
+This logs `"Hello World!"` to the console after a delay of 1s, and so it works
+as expected when we run it... sort of.
+
+But what happens when you want to halt your sleep operation? Or more commonly,
+what happens when you no longer care about the outcome of your sleep operation?
```

Comment:
Since `halt` is not a well known terminology in async/await, It might be easier if you just say `what happens when you no longer care about the outcome of your sleep operation?`